docs(content-releases): add frontend docs (#19391)
strapi/strapi: docs/docs/docs/01-core/content-releases/02-frontend/01-list-view.mdx

Reviewer: We call them pages in the codebase so I think we should be consistent and refer to them as 'pages' everywhere and not 'views'.

---
title: Releases list view
tags:
  - content-releases
  - tech design
---

### Overview

> The list-view provides a comprehensive display of all available content releases. Organized into two tabs, it efficiently categorizes content into pending releases (those not yet published) and completed releases (already published).

Reviewer: ```suggestion
The releases page provides a comprehensive display of all available content releases. Content is organized into two tabs: 'pending' (releases not yet published) and 'done' (releases already published).
```

#### New Release Creation:

> The creation of a new release is made simple with the inclusion of a dedicated button. When activated, this button opens a modal housing a user-friendly form. Importantly, access to this creation feature is contingent on the user holding the necessary permission, specifically `plugin::content-releases.create` otherwise the button is disabled. And for validation the release name needs to be unique and not empty.

Reviewer: ```suggestion
If a user has the permissions to create a release a 'New release' button will be present in the header. Clicking the button opens a form requiring a unique release name. 
```

I believe the button is hidden when permissions are not granted, but I could be wrong...

#### License limits

> A notification is presented above the releases list if, upon page load or after a release creation, the user reaches the license limit. Additionally, the "New release" button is automatically disabled to prevent the creation of any new releases under these circumstances.

Reviewer: ```suggestion
If the user reaches the limit of maximum pending releases defined on their license then the 'New release' button is disabled and a banner is displayed to inform the user. 
```
